sunpy/map/mapbase.py

```diff
@@ -2584,6 +2587,74 @@ def _check_axes(self, axes, warn_different_wcs=False):
 
         return axes
 
+    def reproject_to(self, target_wcs, algorithm='interpolation', **reproject_args):
+        """
+        Reproject the map to a different WCS
+
+        .. note::
+            This method requires the optional package `reproject` to be installed.
+
+        Parameters
+        ----------
+        target_wcs : `dict` or `~astropy.wcs.WCS`
+            The destination FITS WCS header or WCS instance
+        algorithm : str
+            One of the supported `reproject` algorithms (see below)
+
+        Returns
+        -------
+        outmap : `~sunpy.map.GenericMap`
+            The reprojected map
+
+        Notes
+        -----
+        The reprojected map does not preserve any metadata beyond the WCS-associated
+        metadata.
+
+        The supported `reproject` algorithms are:
+
+        * 'interpolation' for :func:`~reproject.reproject_interp`
+        * 'adaptive' for :func:`~reproject.reproject_adaptive`
+        * 'exact' for :func:`~reproject.reproject_exact`
+
+        See the respective documentation for these functions for additional keyword
+        arguments that are allowed.
+
+        .. minigallery:: sunpy.map.GenericMap.reproject_to
+        """
+        try:
+            import reproject
+        except ImportError:
+            raise ImportError("This method requires the optional package `reproject`.")
+
+        if not isinstance(target_wcs, astropy.wcs.WCS):
+            target_wcs = astropy.wcs.WCS(target_wcs)
+
+        # Select the desired reprojection algorithm
+        functions = {'interpolation': reproject.reproject_interp,
+                     'adaptive': reproject.reproject_adaptive,
+                     'exact': reproject.reproject_exact}
+        if algorithm not in functions:
+            raise ValueError(f"The specified algorithm must be one of: {list(functions.keys())}")
+        func = functions[algorithm]
+
+        # Block the `return_footpoint` keyword argument from being passed on
+        if reproject_args.get('return_footprint', False):
+            log.warn("This method does not return the footprint, so `return_footprint` is ignored.")
+        reproject_args['return_footprint'] = False
+
+        # reproject does not automatically grab the array shape from the WCS instance
+        if target_wcs.array_shape is not None:
+            reproject_args.setdefault('shape_out', target_wcs.array_shape)
+
+        # Reproject the array
+        output_array = func(self, target_wcs, **reproject_args)
+
+        # Create and return a new GenericMap
+        outmap = GenericMap(output_array, target_wcs.to_header(),
+                            plot_settings=self.plot_settings)
```

Comment:
hmmm this is a little quirk between ndcube and map. This makes more sense for map, but ndcube does `type(self)()`